### PR TITLE
MM-19870 - reset at-mention selection

### PR DESCRIPTION
**Pull request by ali-farooq0**
#### Summary
Reset at-mention selection

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-19870

#### Screenshots
demo

### CODE REVIEW

File: components/suggestion/suggestion_box.test.jsx [hunk 1]
```diff
@@ -127,4 +128,35 @@ describe('components/SuggestionBox', () => {
...
+                {...props}
+            />
+        );
+        const instance = wrapper.instance();
+
+        expect(wrapper.state('selection')).toEqual('');
+
+        instance.nonDebouncedPretextChanged('hello world @');
+        expect(wrapper.state('selection')).toEqual('@other');
+
```
Comment (jespino) [inline components/suggestion/suggestion_box.test.jsx]: Maybe wen can be more exhaustive here with the test adding:

```suggestion

        instance.nonDebouncedPretextChanged('hello world @u');
        expect(wrapper.state('selection')).toEqual('@user');
        
        instance.nonDebouncedPretextChanged('hello world @');
        expect(wrapper.state('selection')).toEqual('@user');

```

Comment (ali-farooq0) [reply components/suggestion/suggestion_box.test.jsx]: Great suggestion, updated test 👍

**Comment by ali-farooq0** [comment]
Poking @Willyfrog for reveiew :)

**Comment by ali-farooq0** [comment]
Hey @stevemudie adding you on this PR for QA review, feel free to assign someone else from the QA side

**Comment by ali-farooq0** [comment]
just an FYI, this is for 5.19 so feel free to look at other 5.18 PRs first.

**Comment by ali-farooq0** [comment]
@stevemudie much appreciated!

**Pull request merged**